amoro-ams/src/main/java/org/apache/amoro/server/dashboard/MixedAndIcebergTableDescriptor.java

```diff
@@ -505,29 +507,32 @@ public List<ConsumerInfo> getTableConsumerInfos(AmoroTable<?> amoroTable) {
   public Pair<List<OptimizingProcessInfo>, Integer> getOptimizingProcessesInfo(
       AmoroTable<?> amoroTable, String type, ProcessStatus status, int limit, int offset) {
     TableIdentifier tableIdentifier = amoroTable.id();
-    List<OptimizingProcessMeta> processMetaList =
-        getAs(
-            OptimizingMapper.class,
-            mapper ->
-                mapper.selectOptimizingProcesses(
-                    tableIdentifier.getCatalog(),
-                    tableIdentifier.getDatabase(),
-                    tableIdentifier.getTableName()));
-
-    processMetaList =
-        processMetaList.stream()
-            .filter(
-                p ->
-                    StringUtils.isBlank(type)
-                        || type.equalsIgnoreCase(p.getOptimizingType().getStatus().displayValue()))
-            .filter(p -> status == null || status.name().equalsIgnoreCase(p.getStatus().name()))
-            .collect(Collectors.toList());
-
-    int total = processMetaList.size();
-    processMetaList =
-        processMetaList.stream().skip(offset).limit(limit).collect(Collectors.toList());
-    if (CollectionUtils.isEmpty(processMetaList)) {
-      return Pair.of(Collections.emptyList(), 0);
+    int total = 0;
+    // page helper is 1-based
+    int pageNumber = (offset / limit) + 1;
+    List<OptimizingProcessMeta> processMetaList = Collections.emptyList();
+    try (Page<?> ignored = PageHelper.startPage(pageNumber, limit, true)) {
+      processMetaList =
+          getAs(
+              OptimizingMapper.class,
+              mapper ->
+                  mapper.selectOptimizingProcesses(
```

Comment:
need to pass the `type` into the `selectOptimizingProcesses`

Reply:
Yes, we have supported filter type and status on the web.
We may need to support them here.

Reply:
fixed